lib/morse_translator.rb

```diff
@@ -38,6 +38,41 @@ def initialize
                     "9" => "----.",
                     "0" => "-----",
                     " " => " "}
+    @morse_string = ""
+
+  end
+
+
+  def compare_units(element, letter)
+    if element == letter
+      @morse_string += @dictionary[letter]
+    end
+  end
+
+
+  def process_units(array_in)
+    letters = ('a'..'z').to_a
+    [1,2,3,4,5,6,7,8,9,0].to_a.each do |number|
```

Comment:
Here, you are calling #to_a on an array - I think this is a bit unnecessary.  It also seems like a lot of this method is recreating a 'dictionary' type object which we give you above.